### ISSUE TITLE
Lostfilm.tv changed, plugin starts to produce only the errors

### ISSUE THREAD

**Issue by Jabberwacky** [anteo/plugin.video.lostfilm.tv#19]
Starting from today 30.1.2017 plugin on my Raspberry 3 stoped to work at all. As I can see, Lostfilm had changed their site. So, is it a time to change something in a plugin? 
This is the error codes:

ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.ValueError'>
                                            Error Contents: need more than 0 values to unpack
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/addon.py", line 12, in <module>
                                                run_plugin()
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/resources/lib/support/common.py", line 348, in run_plugin
                                                plugin.run()
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/resources/lib/xbmcswift2/plugin.py", line 342, in run
                                                items = self._dispatch(self.request.path)
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/resources/lib/xbmcswift2/plugin.py", line 303, in _dispatch
                                                listitems = view_func(**items)
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/resources/lib/lostfilm/routes.py", line 116, in index
                                                episodes = scraper.browse_episodes(skip)
                                              File "/home/osmc/.kodi/addons/plugin.video.lostfilm.tv/resources/lib/lostfilm/scraper.py", line 282, in browse_episodes
                                                episode_titles, original_titles = zip(*[parse_title(t) for t in titles])
                                            ValueError: need more than 0 values to unpack
                                            -->End of Python script error report<--
20:42:41 124.985718 T:1956828080   ERROR: GetDirectory - Error getting plugin://plugin.video.lostfilm.tv/
20:42:41 124.987030 T:1956828080   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.lostfilm.tv/) failed

**Comment by Ksantor** [comment]
I have investigated the new lostfilm site yesterday, and while it is possible to tweak the plugin to be used with new authentication and series list, I saw no way to get the actual download link for an episode.

The page (popup) with download links is opened with javascript "onclick" event (haven't found the definition of the function). The url of this popup have 6 params:
* series id (pretty easy to get it)
* season number (no problems here)
* episode number (also no problems)
* user id (available on the profile page, just have to get it and cache)
* some kind of MD5 hash (a mystery)
* last parameter is always 1 (at least for me it is)

So basically, if one will figure out what this mysterious MD5 hash is, then current plugin can be updated.

Update: Just found the way to get download links. So for now I see nothing that could stop this plugin from being updated. Maybe will try to do it myself, though, I'm not a python specialist.

**Comment by Jabberwacky** [comment]
Ksantor, I made those changes, which are fixing underscores, but logs still have showed above errors. There is something wrong with browse_episodes subprogram. As I was able to understand, it can't load nothing from site.

**Comment by Jabberwacky** [comment]
Underscores were fixed by myself in a stupid easy manner, previously, I just changed faulty episodes IDs to the 191 ID in the scraper.py file. Just added two strings of python code. That was the top of my knowledges of this language. After approximately two weeks there was four-five IDs in a "database":) [Tommixoft](https://github.com/Tommixoft) had saw the root of problem. And fixed it totally. But now IMHO it is a global trouble.

**Comment by Ksantor** [comment]
@Jabberwacky this is due to the complete redesign of the LostFilm.tv site. Unfortunately the underscore fix is useless now, and much more work has to be done to get the plugin working again:

* update the authentication flow
* update the series and episodes list parser
* update the series library parser

**Comment by Jabberwacky** [comment]
Yes... I Lost my Films...

**Comment by Ksantor** [comment]
I'm not sure, if @anteo is interested in maintaining/updating this plugin further, but I'll try to fix it myself, though, can't promise a fast result.

**Comment by Jabberwacky** [comment]
That would be perfect :)))! Unfortunately, I know a little about web programming. Better to say know nothing. So I can't help.

P.S.
January 2017
anteo has no activity yet for this period.
May - December 2016
anteo had no activity during this period.

Sure, noble Anteo started to work on more interesting projects... So, dear Ksantor, you are - the only ours hope :)

**Comment by Tommixoft** [comment]
Lostfilm.tv changed their website so this plugin is doomed or have to be all regex and url rewritten in addon.

With Kodi addons is hard to develop cause you cant debug it.

**Comment by Jabberwacky** [comment]
It's a bunch of work to rebuild all in an addon... Nobody knows this plugin as its creator. I guess, if Anteo will not take part in this work, no one will load it on his shoulders. LostFilm RIP

**Comment by Ksantor** [comment]
Guys, I'll do my best, because I used this plugin heavily myself and don't want to let it go just like that.

@Tommixoft eveything is not so bad as it looks.

**Comment by Tommixoft** [comment]
@Ksantor - im .NET, php programmer i dont write python, but anyways when i look at source - he almost never use TRY, he never check is list empty or null. he just uses variables not even thinking that it may be empty or some error may happen. That's why underscore broke plugin. Its minor problem but broke whole plugin :) I know now underscore is least of problem, but first undersscore broke plugin, well datetime format broke it, then underscore and now web page redesign :D

**Comment by Jabberwacky** [comment]
It is my style of work ;) 
This addon is splitted aparts pretty well. The only way to debug everything - is to see the logs. Annoying. I am a, commonly, VB.NET 'programmer' VisualStudio - is the only programming environment for me.

**Comment by Ksantor** [comment]
Just to keep everybody informed:

Made a [fork](https://github.com/Ksantor/plugin.video.lostfilm.tv). Nothing important to see there yet, but will keep working on it!

**Comment by Kirrrilka** [comment]
почему все общаются на английском про сайт, который переводит фильмы на Русский?

**Comment by Tommixoft** [comment]
Because for example i do understand russian good but i cant write in russian :)

**Comment by Jabberwacky** [comment]
:)))) вай нот? Кому-то может оказаться интересной информация по этому аддону, но он может не знать русского. А может смотрит знакомые сериалы на русском, чтобы его изучить? Я просто забыл о сути самого аддона и был огорчен его сбоем и думал как его починить. Все что у меня получалось ранее с этим аддоном - вырезать часть таймстемпа, чтобы не было глюков с изменившимся таймстемпом файлов и заменять сбойнутые имена на гарантированно нормальное, когда возникла ошибка с подчеркиванием. Оказалось, что предстоит работка на порядки более сложная. О том, что показывал аддон, я совсем забыл.

**Comment by Ksantor** [comment]
Looks like I made some real progress!

Will test it myself in "real life". In case of success will post `*.zip` links and a README in my repo.

**Comment by Ksantor** [comment]
Uploaded a first version in my repo. You can check it out and open new issues, if you'll find any bugs or encounter any problems.

**Comment by Tommixoft** [comment]
@Ksantor sad but doesnt work. and your repo doesnt have ISSUES tab. installed on 2 systems when i go to plugin it shows 0 items. no menu, nothing... like im in empty folder.

**Comment by Tommixoft** [comment]
```
8:41:26 T:9948  NOTICE: [plugin.video.lostfilm.tv] Starting LostFilm.TV library update...
18:41:26 T:9948  NOTICE: [support.library] Starting library sync...
18:41:26 T:9948  NOTICE: [support.library] Library sync finished (0 file(s) created, 0 added, 0 updated, 0 removed)
18:41:26 T:9948  NOTICE: [plugin.video.lostfilm.tv] LostFilm.TV library update finished.
18:41:26 T:9948  NOTICE: [plugin.video.lostfilm.tv] Scheduling next library update at 2017-02-06 06:41:26.325000
18:42:05 T:6724  NOTICE: [xbmcswift2] Request for "/" matches rule for function "index"
18:42:05 T:6724  NOTICE: [lostfilm.network_request] Authorization started...
18:42:05 T:6724  NOTICE: [lostfilm.network_request] Fetching URL https://www.lostfilm.tv/ajaxik.php with params None started...
18:42:05 T:6724   ERROR: C:\Users\Tomas\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\packages\urllib3\util\ssl_.py:334: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              SNIMissingWarning
18:42:05 T:6724   ERROR: C:\Users\Tomas\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\packages\urllib3\util\ssl_.py:132: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                                              InsecurePlatformWarning
18:42:07 T:6724  NOTICE: [lostfilm.network_request] Fetching URL https://www.lostfilm.tv/ajaxik.php with params None taken: 1.938154 seconds
18:42:07 T:6724  NOTICE: [lostfilm.network_request] Authorization taken: 1.938439 seconds
18:42:07 T:6724  NOTICE: [lostfilm.network_request] Fetching URL https://www.lostfilm.tv/my/type_1 with params None started...
18:42:07 T:6724  NOTICE: [lostfilm.network_request] Fetching URL https://www.lostfilm.tv/my/type_1 with params None taken: 0.065396 seconds
```

**Comment by Ksantor** [comment]
@Tommixoft, looks like some SSL error. Previous implementation used `http` for authentication. I've changed the urls to `https` (by the way, be careful, lostfilm.tv web page doesn't switch to https by default).

If you don't mind sending your credentials for lostfilm.tv account over http, you can try to fix it youself:
in `\AppData\Roaming\Kodi\addons\plugin.video.lostfilm.tv\resources\lib\lostfilm\network_request.py` change

```
20: BASE_URL = "https://www.lostfilm.tv"
21: POST_URL = "https://www.lostfilm.tv/ajaxik.php"
```
to:

```
20: BASE_URL = "http://www.lostfilm.tv"
21: POST_URL = "http://www.lostfilm.tv/ajaxik.php"
```
Should help.

Meanwile, I'll try to understand, what is the problem with SSL.

P.S. Added "Issues" page to my repo.

**Comment by Tommixoft** [comment]
Thanks. Will try. 
Maybe you're using linux? windows is not so easy with ssl.  Maybe you can somehow bypass SSL veryfication and so on.

**Comment by Tommixoft** [comment]
@Ksantor Not working anyways...log became even smaller. Plugin still shows 0 entries, no even static menu like "My Library".

```
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Authorization started...
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Fetching URL http://www.lostfilm.tv/ajaxik.php with params None started...
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Fetching URL http://www.lostfilm.tv/ajaxik.php with params None taken: 0.079945 seconds
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Authorization taken: 0.080223 seconds
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Fetching URL http://www.lostfilm.tv/my/type_1 with params None started...
20:18:47 T:5888  NOTICE: [lostfilm.network_request] Fetching URL http://www.lostfilm.tv/my/type_1 with params None taken: 0.051973 seconds
```

**Comment by capris** [comment]
Hello all, for me everything is working :), thank you, one thing in old plugin and website you need to login with nickname, now it is email :), first tried with nickname and i had empty list :)

**Comment by Tommixoft** [comment]
@capris  It always was email, atleast im using email year already. What OS you're running?

**Comment by capris** [comment]
newest, krypton on android tv.
i was using nicname till today :)
and did you marked series as favorite in the web as said in the readme ?

**Comment by Jabberwacky** [comment]
Dear guys! Look, what I found on the XBMC.RU forum. It is a temporary solution.
http://xbmc.ru/forum/showthread.php?t=12469&highlight=lostfilm&page=18

eczo
Junior Member
 
Регистрация: 05.02.2017
Сообщений: 1
По умолчанию
Всем привет

Нашел временное решение:
В plugin.video.lostfilm.tv/resources/lib/lostfilm/scraper.py
поправил переменную:
BASE_URL = "http://www.lostfilm.tv"
на
BASE_URL = "http://old.lostfilm.tv"
Плагин заработал, со старой регистрацией.
Не знаю, сколько еще проживет old.lostfilm.tv, но адаптация под новый сайт, это не просто замена url на новый, парсер также нужно править.
P.s. у меня работает на Ubuntu 16.04 + kodi 17 и Xiaomi Mi Box Android TV 6 + kodi 17
Последний раз редактировалось eczo; Вчера в 13:24.

**Comment by Ksantor** [comment]
@Jabberwacky just found the same thing about an hour ago while was investigating the new lostfilm site. Really looks like a valid solution, though, yea, who knows for how long the old site will be active.

**Comment by omikhailovskii** [comment]
I have started porting of the the original addon to work with the new Lostfilm interface. Please, find the project via the link: https://github.com/winnipeg8/plugin.video.lostfilm.tv. Looks like it is workable. Basically, the only changes I have done are in "resources/lib/lostfilm/" folder. The biggest issues so far are: 1) their new authentication protocol (I am still using the old one); 2) details information parsing. It would be really cool, if someone could help me with these problems. Also, don't get mad on my code changes - I am not a professional:)

**Comment by Tommixoft** [comment]
@winnipeg8 original plugin was written poorly. I more like @Ksantor version. Also i started to create new features to Ksantor version. Of course it's SUPER basic now. I think it will never have Library support, which anyways was broken in original plugin. This new version will be (at least so far) online, no caching of series, episodes. Full read of everything from website (favorites for example)

**Comment by Ksantor** [comment]
@winnipeg8 since I'm also already started a total refactoring of this plugin, you can check my profile. I think I'll create a new repo for my version of the plugin.

Right now there is only a fixed (fast and dirty fix) of the original plugin.

In the new version I'll be using updated xbmcswift2 library (kodiswift), another html parser - parsedom, and couple more libraries as dependencies instead of bundles inside the plugin itself.

Plus, probably, will make a kodi repository to support OTA updates.

As @Tommixoft mentioned, no local library for now, though local-web sync alongside with proper caching is in my plans.

I think it will be great, if we will join forces, thus making a good and maintainable/extendable plugin.

**Comment by omikhailovskii** [comment]
@Ksantor so I loaned your authorization procedure (I was kinda on your way but didn't know how to complete it). Now the plugin should be fully compatible with the new LF interface, and also has the old @anteo 's original (info fetching is still to be implemented) features. In a few words, I store the database of series_ids vs web_ids, i.e. 125: Its_Always_Sunny_in_Philadelphia in a separate file and update it when a new series premiere airs. Thus, it allows load the pages quicker than it took.

I agree with you guys, that probably the plugin should be rewritten. For now I see my fix as a huge crutch.

I'll take a look on the weekend when I have time. Meanwhile, since @Ksantor you know the organization of the original plugin it should be easier for you to notice all the changes (it is really just 3 files) that I have done and it should be relatively easy to integrate them (such as, new episodes fetching and all series fetching).

**Comment by Tommixoft** [comment]
@winnipeg8  your version don't even start on my pc :(

**Comment by Kirrrilka** [comment]
By @Tommixoft 
 I think it will never have Library support, which anyways was broken in original plugin.

It is working. It had small issue with series ids with underscore. But I fixed in my TV. I can share the fix if you want. Actually Library is very useful, it highlites new episodes in series. 

Currently I am using old plugin with changed BASE_URL to old.lostfilm.com
It have several my fixes and now it very functional. It even displays all series list (which also was not so functional last time)

**Comment by Tommixoft** [comment]
@Kirrrilka  for me library (kodi native) only worked for 1 tv show. Whatever i added more - always showed 1, and that's it :) I agree that original plugin is very useful. 
But now with @Ksantor version i can a bit udnerstand how t works, what it does. so i can now develope by myself (im not python friend or nether i know KODI api). 
Now im trying to keep @Ksantor code intact, but i would like to restructure things, optimize a bit authorization, so i may endup with my own version, for my private use, i will upload to github to share, as i did.
Library support is nice feature, i agree. But i think it will be implemented last and by smarter someone than me :)

**Comment by omikhailovskii** [comment]
@Tommixoft could you share your kodi.log to know what throws errors? As a blind guess, have you cleared 'userdata' folder for the plugin? Currently, I tested it on Ubuntu and OSMC distros.

**Comment by Tommixoft** [comment]
@winnipeg8  it's hard for me to get log from that pc. dont have keyboard and so on, but it breaks because of some `_new()_ `requires 12 parameters but 11 got.

So you should look at what INDEX route loads and that part crashes. This looks like syntax problem not influenced by any userdata stuff. 

if you still need i can later when im back from work look exactly for log.

**Comment by Allin78** [comment]
[scraper.zip](https://github.com/anteo/plugin.video.lostfilm.tv/files/983081/scraper.zip)
Вот вам временное решение. Положить в .kodi/addons/plugin.video.lostfilm.tv/resources/lib/lostfilm с заменой. Старый функционал !не урезан! Спасибо заносить пользователю shprota с форума xbmc.ru/forum

**Comment by Tommixoft** [comment]
Why you need temporal solutions when i've rewrite this plugin 3 months ago and it works great.
https://github.com/Tommixoft/plugin.video.lostfilm.tv/releases

**Comment by Allin78** [comment]
Integration with kodi library are still alive in your plugin?

8 мая 2017 г. 13:22 пользователь "Tomas" <notifications@github.com> написал:

**Comment by Tommixoft** [comment]
Hi Alin78 - no, It's not so critical, so i did not implement it. Also in original plugin it was buggy and did not work correct. Only with 1 tv show and others never worked for me.

I will try to implement it, but...dont know when. My priority is to implement ability to add to favorites from plugin. Im very new to KODI and i dont know how to call action from context menu.

I think having stable plugin is much more important than features that is not implemented. My GIT have no issues added so do not work on it because i dont know how many people uses it.

**Comment by Allin78** [comment]
Original plugin from anteo worked for me fine. And integration of new
series to default kodi library is awesome feature and yes it worked too.
Edit of two files is not to hard for me

8 мая 2017 г. 15:07 пользователь "Tomas" <notifications@github.com> написал:

**Comment by Tommixoft** [comment]
Strange for me never worked. I added 1 tv show it was OK and after i added 2..3... those never appeared in library. and i couldn't delete the first one.
My plugin have few features original did not had. Like "newest tv shows", "Best rated tv shows". Sometimes good if you don't know what to look :)
I'll try to implement library support... will be good lesson for me. And thanks for writing english. I understand russian, but i don't know how to write in russian.
Russians have great websites! :D where else you find almost all tv shows with professional translation and free of charge :)

**Comment by Allin78** [comment]
[hideme.zip](https://github.com/anteo/plugin.video.lostfilm.tv/files/984868/hideme.zip)
Sorry! I forgot this - replace file at this path ".kodi/addons/plugin.video.lostfilm.tv/resources/lib/support" too. Just remember: both of changed files working only with registation on OLD site. Cookie's magic :)

**Comment by capris** [comment]
hey if someone is still using the plugin, something strange started to happened, about a week it is not possible to list the series and of course to watch them, list is just empty. and if i try to use some other gateway like other internet provider - i can use plugin normally but only for one movie, after that it locks again, maybe someone know why is this happening, some kind of blocking by ip

**Comment by Tommixoft** [comment]
you cna use my plugin https://github.com/Tommixoft/plugin.video.lostfilm.tv/releases it have fix for this, but it's not 100% possible to 100% of the time to fix. Lostfilm added captcha protection to login form and script can not fill captcha. In my plugin page you can see how to hack this by login to lostfilm from your pc. 
My plugin only logs in once, so it will not have problem with captcha, i think this si easy to fix in this plugin too. Mor einfo about this https://github.com/Tommixoft/plugin.video.lostfilm.tv/wiki/Plugin-no-longer-works%3F

**Comment by capris** [comment]
thank you, it looks like it is working on one device, later will check on others.
